### PR TITLE
COMMERCE-11516 Adding onAfterRemove listener for AccountEntry to unli…

### PR DESCRIPTION
**Pull request by MahmoudAzzam**
Jira : https://issues.liferay.com/browse/COMMERCE-11516

### CODE REVIEW

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/model/listener/AccountEntryModelListener.java [hunk 1]
```diff
@@ -35,6 +41,29 @@
...
 public class AccountEntryModelListener extends BaseModelListener<AccountEntry> {
 
+	@Override
+	public void onAfterRemove(AccountEntry accountEntry) {
+		List<CommerceCatalog> commerceCatalogs =
+			_commerceCatalogLocalService.getCommerceCatalogs(
+				accountEntry.getCompanyId(), false);
+
+		for (CommerceCatalog commerceCatalog : commerceCatalogs) {
+			commerceCatalog.setAccountEntryId(0);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/model/listener/AccountEntryModelListener.java:51]: You should update only the catalogs that are linked to the deleted account, not every catalog of the company

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/model/listener/AccountEntryModelListener.java [hunk 2]
```diff
@@ -35,6 +41,29 @@
...
+
+			_commerceCatalogLocalService.updateCommerceCatalog(commerceCatalog);
+		}
+
+		List<CommerceChannel> commerceChannels =
+			_commerceChannelLocalService.getCommerceChannels(
+				accountEntry.getCompanyId());
+
+		for (CommerceChannel commerceChannel : commerceChannels) {
+			commerceChannel.setAccountEntryId(0);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/model/listener/AccountEntryModelListener.java:61]: same here

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.

Thank you!

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
Hi @MahmoudAzzam could you tell me why your last commit include another ticket here? Does it make sense? Thanks in advance!

cc/ @victoroscuro

**Comment by MahmoudAzzam** [comment]
> Hi @MahmoudAzzam could you tell me why your last commit `COMMERCE-10421 baseLine` include another ticket here? Does it make sense? Thanks in advance!
> 
> cc/ @victoroscuro

Hi Jayda, sorry it was a mistake. It is fixed now

**Comment by jaydawu** [comment]
QA passed, add automation and resent on https://github.com/liferay-commerce/liferay-portal/pull/3905

**Comment by jaydawu** [comment]
ci:close

**Pull request closed without merging**